x-pack/plugins/security_solution/public/common/hooks/use_upgrade_security_packages.ts

```diff
@@ -9,25 +9,53 @@ import { useEffect } from 'react';
 import type { HttpFetchOptions, HttpStart } from '@kbn/core/public';
 import type { BulkInstallPackagesResponse } from '@kbn/fleet-plugin/common';
 import { epmRouteService } from '@kbn/fleet-plugin/common';
-import { useKibana } from '../lib/kibana';
+import type { InstallPackageResponse } from '@kbn/fleet-plugin/common/types';
+import { KibanaServices, useKibana } from '../lib/kibana';
 import { useUserPrivileges } from '../components/user_privileges';
 
 /**
  * Requests that the endpoint and security_detection_engine package be upgraded to the latest version
  *
  * @param http an http client for sending the request
  * @param options an object containing options for the request
+ * @param prebuiltRulesPackageVersion specific version of the prebuilt rules package to install
  */
 const sendUpgradeSecurityPackages = async (
   http: HttpStart,
-  options: HttpFetchOptions = {}
-): Promise<BulkInstallPackagesResponse> => {
-  return http.post<BulkInstallPackagesResponse>(epmRouteService.getBulkInstallPath(), {
-    ...options,
-    body: JSON.stringify({
-      packages: ['endpoint', 'security_detection_engine'],
-    }),
-  });
+  options: HttpFetchOptions = {},
+  prebuiltRulesPackageVersion?: string
+): Promise<Array<PromiseSettledResult<BulkInstallPackagesResponse | InstallPackageResponse>>> => {
+  const packages = ['endpoint', 'security_detection_engine'];
+  const requests: Array<Promise<InstallPackageResponse | BulkInstallPackagesResponse>> = [];
+
+  // If `prebuiltRulesPackageVersion` is provided, try to install that version
+  // Must be done as two separate requests as bulk API doesn't support versions
+  if (prebuiltRulesPackageVersion != null) {
+    packages.splice(packages.indexOf('security_detection_engine'), 1);
+    requests.push(
+      http.post<InstallPackageResponse>(
+        epmRouteService.getInstallPath('security_detection_engine', prebuiltRulesPackageVersion),
+        {
+          ...options,
+          body: JSON.stringify({
+            force: true,
+          }),
+        }
+      )
+    );
+  }
+
+  // Note: if `prerelease:true` option is provided, endpoint package will also be installed as prerelease
+  requests.push(
+    http.post<BulkInstallPackagesResponse>(epmRouteService.getBulkInstallPath(), {
+      ...options,
+      body: JSON.stringify({
+        packages,
+      }),
+    })
+  );
+
+  return Promise.allSettled(requests);
```

Comment:
Nit: We could make it return `void` since the result is not used in `useUpgradeSecurityPackages`.

```ts
Promise.allSettled(requests);
return;
```

Reply:
Addressed in [199c0af](https://github.com/elastic/kibana/pull/148426/commits/199c0af96c08c0b6fa4d66419087a676b35f4851) 👍